src/DotVVM.Framework/ResourceManagement/LocalResourceUrlManager.cs

```diff
@@ -77,20 +116,36 @@ private ILocalResourceLocation TryLoadAlternativeFile(string name, string hash,
                 if (directory != null)
                 {
                     var sourceFile = Path.Combine(directory, name);
-                    if (File.Exists(sourceFile)) return new FileResourceLocation(sourceFile);
+                    if (File.Exists(sourceFile))
+                    {
+                        return new FileResourceLocation(sourceFile);
+                    }
                 }
             }
             return null;
         }
 
-        protected ILocalResourceLocation FindLocation(IResource resource, out string mimeType)
+        protected ILocalResourceLocation FindLocation(
+            IResource resource,
+            int? locationIndex,
+            out string mimeType)
         {
-            var linkResource = resource as ILinkResource;
-            mimeType = linkResource?.MimeType;
-            return linkResource
-                   ?.GetLocations()
-                   ?.OfType<ILocalResourceLocation>()
-                   ?.FirstOrDefault();
+            if (!(resource is ILinkResource link))
+            {
+                mimeType = null;
+                return null;
+            }
+
+            mimeType = link.MimeType;
+            var locations = link.GetLocations().ToArray();
```

Comment:
Why the `ToArray`?

Reply:
I'm assuming the order of locations provided by `GetLocations` is stable. To be frank, I'm not entirely certain whether this assumption holds water.